src/extrema.jl

```diff
@@ -0,0 +1,117 @@
+"""
+BlobLoG stores information about the location of peaks as discovered by `blob_LoG`.
+It has fields:
+
+- location: the location of a peak in the filtered image (a CartesianIndex)
+- σ: the value of σ which lead to the largest `-LoG`-filtered amplitude at this location
+- amplitude: the value of the `-LoG(σ)`-filtered image at the peak
+
+Note that the radius is equal to σ√2.
+
+See also: [`blob_LoG`](@ref).
+"""
+struct BlobLoG{T,S,N}
+    location::CartesianIndex{N}
+    σ::S
+    amplitude::T
+end
+
+"""
+    blob_LoG(img, σscales; edges=(true, false, ...), σshape=(1, ...)) -> Vector{BlobLoG}
+
+Find "blobs" in an N-D image using the negative Lapacian of Gaussians
+with the specifed vector or tuple of σ values. The algorithm searches for places
+where the filtered image (for a particular σ) is at a peak compared to all
+spatially- and σ-adjacent voxels, where σ is `σscales[i] * σshape` for some i.
+By default, `σshape` is an ntuple of 1s.
+
+The optional `edges` argument controls whether peaks on the edges are
+included. `edges` can be `true` or `false`, or a N+1-tuple in which
+the first entry controls whether edge-σ values are eligible to serve
+as peaks, and the remaining N entries control each of the N dimensions
+of `img`.
+
+# Citation:
+
+Lindeberg T (1998), "Feature Detection with Automatic Scale Selection",
+International Journal of Computer Vision, 30(2), 79–116.
+
+See also: [`BlobLoG`](@ref).
+"""
+function blob_LoG(img::AbstractArray{T,N}, σscales::Union{AbstractVector,Tuple};
```

Comment:
Maybe deprecate `AbstractVector` input for `σscales`?

Reply:
I was in initial agreement but actually `σscales` is not guaranteed to be small. It doesn't correspond to, say, array dimensions, instead it will be the size of the first dimension of `img_LoG`. There's no reason that couldn't be 20 or higher, at which point using tuples doesn't make a lot of sense.